### ISSUE TITLE
Conflict in dependency resolution when installing @angular/cdk@16.0.2

### ISSUE THREAD

**Issue by Nomackleo** [angular/components#27203]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

@angular/common 15

### Description

I am experiencing a dependency conflict while trying to install @angular/cdk@16.0.2 in an Angular project. The project currently has the following dependencies:

`"dependencies": {
    "@angular/animations": "^16.1.0-next.2",
    "@angular/common": "^16.1.0-next.2",
    "@angular/compiler": "^16.1.0-next.2",
    "@angular/core": "^16.1.0-next.2",
    "@angular/forms": "^16.1.0-next.2",
    "@angular/platform-browser": "^16.1.0-next.2",
    "@angular/platform-browser-dynamic": "^16.1.0-next.2",
    "@angular/router": "^16.1.0-next.2",
    "rxjs": "~7.8.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.13.0"
}
`
When I try to install @angular/cdk using the command ng add @angular/cdk@16.0.2, I receive the following error:

`The package @angular/cdk@16.0.2 will be installed and executed.
Would you like to proceed? Yes
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: game@0.0.0
npm ERR! Found: @angular/common@16.1.0-next.2
npm ERR! node_modules/@angular/common
npm ERR!   @angular/common@"^16.1.0-next.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/common@"^16.0.0 || ^17.0.0" from @angular/cdk@16.0.2
npm ERR! node_modules/@angular/cdk
npm ERR!   @angular/cdk@"16.0.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.`

### Reproduction

Steps to reproduce:
1. Create an Angular project.
2. Run the command `ng add  @angular/cdk`.
3. Run the command `npm update`.
4. Run the command `npm install`.
5. Run the command `ng add @angular/cdk --legacy-peer-deps`
6. Run the command `ng add @angular/cdk --force`.

### Expected Behavior

I expect @angular/cdk@16.0.2 to be installed without any dependency conflicts, considering the specified versions of the other Angular dependencies.

### Actual Behavior

The installation fails with the error message mentioned above.

### Environment

- Angular: ^16.1.0-next.2
- CDK/Material:   @angular/cdk@"16.0.2
- Browser(s): N/A
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 10

**Comment by mmalerba** [comment]
Since you're using the `next` version of Angular, it probably makes sense to `ng add  @angular/cdk@next` instead.

@josephperrott @devversion do either of you know if `ng add  @angular/cdk` is expected to work when the next version of Angular is installed? The error message states it is looking for `peer @angular/common@"^16.0.0 || ^17.0.0" from @angular/cdk@16.0.2`, and it does feel like that should include `16.1.0-next.*`

**Comment by Nomackleo** [comment]
I already tried using `ng add @angular/cdk@next` but the error continues.

Thank you very much for answering @mmalerba.

**Comment by mmalerba** [comment]
Have you tried deleting the `package-lock.json`? The error message indicates that its not actually using the next version of CDK.

**Comment by Nomackleo** [comment]
@mmalerba I just tried and the error continues.
Do you know what version of angular 16 is stable with `@angular/sdk`?

**Comment by mmalerba** [comment]
`@angular/cdk@latest` should work with `@angular/core@latest`

You can see the current latest versions on npm:
- [@angular/core](https://www.npmjs.com/package/@angular/core?activeTab=versions)
- [@angular/cdk](https://www.npmjs.com/package/@angular/cdk?activeTab=versions)

**Comment by devversion** [comment]
@mmalerba Re your question: I don't think our peer dependency would show that e.g. a stable CDK version is compatible with a next unstable version of the framework- so peer dep seems correct

**Comment by mmalerba** [comment]
Ok, closing this then since it sounds like its working as expected

**Comment by Nomackleo** [comment]
I uninstalled Angular globally. I've tried versions of Angular prior to 16.1.0-next.2, but the error persisted. Then I tried version 16.1.0-next.3 and the error still persists.

**Comment by Nomackleo** [comment]
`Log Files: # npm resolution error report

While resolving: undefined@undefined
Found: @angular/core@16.1.0-next.3
node_modules/@angular/core
  peer @angular/core@"16.1.0-next.3" from @angular/animations@16.1.0-next.3
  node_modules/@angular/animations
    peerOptional @angular/animations@"16.1.0-next.3" from @angular/platform-browser@16.1.0-next.3
    node_modules/@angular/platform-browser
      peer @angular/platform-browser@"16.1.0-next.3" from @angular/forms@16.1.0-next.3
      node_modules/@angular/forms
      peer @angular/platform-browser@"16.1.0-next.3" from @angular/platform-browser-dynamic@16.1.0-next.3
      node_modules/@angular/platform-browser-dynamic
      peer @angular/platform-browser@"16.1.0-next.3" from @angular/router@16.1.0-next.3
      node_modules/@angular/router
  peer @angular/core@"16.1.0-next.3" from @angular/common@16.1.0-next.3
  node_modules/@angular/common
    peer @angular/common@"16.1.0-next.3" from @angular/forms@16.1.0-next.3
    node_modules/@angular/forms
    peer @angular/common@"16.1.0-next.3" from @angular/platform-browser@16.1.0-next.3
    node_modules/@angular/platform-browser
      peer @angular/platform-browser@"16.1.0-next.3" from @angular/forms@16.1.0-next.3
      node_modules/@angular/forms
      peer @angular/platform-browser@"16.1.0-next.3" from @angular/platform-browser-dynamic@16.1.0-next.3
      node_modules/@angular/platform-browser-dynamic
      peer @angular/platform-browser@"16.1.0-next.3" from @angular/router@16.1.0-next.3
      node_modules/@angular/router
    peer @angular/common@"16.1.0-next.3" from @angular/platform-browser-dynamic@16.1.0-next.3
    node_modules/@angular/platform-browser-dynamic
    peer @angular/common@"16.1.0-next.3" from @angular/router@16.1.0-next.3
    node_modules/@angular/router
  peerOptional @angular/core@"16.1.0-next.3" from @angular/compiler@16.1.0-next.3
  node_modules/@angular/compiler
    peer @angular/compiler@"16.1.0-next.3" from @angular/compiler-cli@16.1.0-next.3
    node_modules/@angular/compiler-cli
      peer @angular/compiler-cli@"^16.0.0 || ^16.1.0-next.0" from @angular-devkit/build-angular@16.1.0-next.2
      node_modules/@angular-devkit/build-angular
      peer @angular/compiler-cli@"^16.0.0 || ^16.1.0-next.0" from @ngtools/webpack@16.1.0-next.2
      node_modules/@ngtools/webpack
        @ngtools/webpack@"16.1.0-next.2" from @angular-devkit/build-angular@16.1.0-next.2
        node_modules/@angular-devkit/build-angular
    peer @angular/compiler@"16.1.0-next.3" from @angular/platform-browser-dynamic@16.1.0-next.3
    node_modules/@angular/platform-browser-dynamic
  peer @angular/core@"16.1.0-next.3" from @angular/forms@16.1.0-next.3
  node_modules/@angular/forms
  peer @angular/core@"16.1.0-next.3" from @angular/platform-browser@16.1.0-next.3
  node_modules/@angular/platform-browser
    peer @angular/platform-browser@"16.1.0-next.3" from @angular/forms@16.1.0-next.3
    node_modules/@angular/forms
    peer @angular/platform-browser@"16.1.0-next.3" from @angular/platform-browser-dynamic@16.1.0-next.3
    node_modules/@angular/platform-browser-dynamic
    peer @angular/platform-browser@"16.1.0-next.3" from @angular/router@16.1.0-next.3
    node_modules/@angular/router
  peer @angular/core@"16.1.0-next.3" from @angular/platform-browser-dynamic@16.1.0-next.3
  node_modules/@angular/platform-browser-dynamic
  peer @angular/core@"16.1.0-next.3" from @angular/router@16.1.0-next.3
  node_modules/@angular/router

Could not resolve dependency:
@angular/cdk@"16.0.3" from the root project

Conflicting peer dependency: @angular/core@16.0.4
node_modules/@angular/core
  peer @angular/core@"^16.0.0 || ^17.0.0" from @angular/cdk@16.0.3
  node_modules/@angular/cdk
    @angular/cdk@"16.0.3" from the root project

Fix the upstream dependency conflict, or retry
this command with --force or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.`

**Comment by Nomackleo** [comment]
Apologies for the inconvenience caused and resolution of the compatibility issue.
The specific error I faced was related to a conflict in dependency resolution while attempting to install the "@angular/cdk@16.0.2" package. I was able to identify the root cause of the problem.

I proceeded to upgrade Node.js to version 20.2.0. Resolving the compatibility conflict with the dependencies, allowing me to successfully install the "@angular/cdk" package without any further issues.
